packages/taco/src/conditions/context/context.ts

```diff
@@ -112,7 +112,7 @@ export class ConditionContext {
   }
 
   private isContextParameter(param: unknown): boolean {
-    return typeof param === 'string' && param.startsWith(CONTEXT_PARAM_PREFIX);
+    return !!String(param).match(CONTEXT_PARAM_REGEXP);
```

Comment:
I was thinking about this a bit more.

So currently the zod set up for parameters/values is:
```
export const ContextParamSchema = z.string().regex(CONTEXT_PARAM_REGEXP);
export const ParamSchema = z.unknown();
export const ParamOrContextParamSchema = z.union([
  ParamSchema,
  ContextParamSchema,
]);
```

Since this is a union that contains z.unknown() I suspect that we don't get any real validation of context parameters at the zod level of validation.

So it begs the question, whether in this method, you should do the same thing that @cygnusv does in his pending PR - https://github.com/nucypher/nucypher/pull/3331/files#diff-f364280e61c1cdd8b5991c2c34c40e0b85cd87f172f15bd7c3454dd30919fad6R79

Which is:
- check if string, and starts with "colon" (call this check (a))
    - if (a) passes and regex validates then return true
    - if (a) passes but then validating with the regex fails, then throw an error - this is basically the case of an incorrect context variable  
- if (a) fails then return false - non-context variable parameter value.

cc @cygnusv .

Reply:
I confirmed this is an issue, i.e. `zod` is not correctly discriminating between a plain string with a `:` and a context parameter. When a string contains both `:` and an illegal context parameter character (like `&`), `zod` interprets it as a string.

I've added a more thorough validation based on your suggestions. I also made the related parameter schema less-permissive because it contributed to this issue - `z.unknown()` type matches any type and value.